application/F3DStarter.cxx

```diff
@@ -417,12 +417,65 @@ class F3DStarter::F3DInternals
       std::to_string(maxNumberingAttempts) + " attempts");
   }
 
+  static bool PatternMatched(const std::string& source, const std::string& matchType,
+    const std::string& match, const std::string& inputFile)
+  {
+    static std::set<std::string> erroredPatterns;
+    if (matchType == "exact")
+    {
+      return match == inputFile;
+    }
+    else
+    {
+      try
+      {
+        std::regex re(matchType == "glob"
+            ? f3d::utils::globToRegex(
+                match, f3d::utils::globContainsGlobstar(match), fs::path::preferred_separator)
+            : match,
+          std::regex_constants::icase);
+        return std::regex_match(inputFile, re);
+      }
+      catch (const f3d::utils::glob_exception& ex)
+      {
+        if (auto [it, inserted] = erroredPatterns.emplace(match); inserted)
+        {
+          f3d::log::error("There was an error in the config ", source, " for glob pattern `", match,
+            "`: ", ex.what());
+        }
+        return false;
+      }
+      catch (const std::regex_error& ex)
+      {
+        if (auto [it, inserted] = erroredPatterns.emplace(match); inserted)
+        {
+          f3d::log::error("There was an error in the config ", source, " for ", matchType,
+            " pattern `", match, "`: ", ex.what());
+        }
+        return false;
+      }
+    }
+  }
+
   static void PrintLoggingMap(const std::map<std::string, log_entry_t>& loggingMap, char sep)
   {
     for (const auto& [key, tuple] : loggingMap)
     {
-      const auto& [bindStr, source, pattern, commands] = tuple;
-      std::string origin = source.empty() ? pattern : std::string(source) + ":`" + pattern + "`";
+      const auto& [bindStr, source, matchType, match, commands] = tuple;
+      std::string origin;
+      if (source.empty())
+      {
+        origin = match;
+      }
+      else
+      {
+        origin = source;
+        origin += ":`";
+        origin += match;
+        origin += "` (";
+        origin += matchType;
+        origin += ")";
```

Comment:
```suggestion
        origin = source + ":`" + match + "` (" + matchType +  ")";
```
?

Reply:
if the goal is to keep in one line it'd be better to use chained `.append()` calls than to bypass `clang-tidy` (even tho it's not critical, the warning is sensible)
example: https://github.com/f3d-app/f3d/blob/c117e5b76ed366eb11b313b34529f8e7d6e2c663/library/src/interactor_impl.cxx#L820

now is a bunch of `.append()` on a single line more readable than a bunch of `+=` on multiple lines is another question...

Reply:
fine to keep as is then ^^.
You can add a comment above mentionning using std::format once c++20 is supported :)